src/Void.Core/BufferMessages.fs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Void.Core
+
+type BufferMessage = inherit Message
+
+type BufferEnvelopeMessage<'TBufferMessage when 'TBufferMessage :> BufferMessage> =
+    {
+        BufferId : int
+        Message : 'TBufferMessage
+    }
+    interface EnvelopeMessage<'TBufferMessage>

```

Comment:
I think I can see it. It's just providing those things that are common to a `BufferMessage` whereas the BM itself has specifics. Cool!